File: VB/XtraReport1.vb
Imports System
Imports System.Drawing
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraReports.UI.PivotGrid

Namespace WindowsFormsApplication1

    Public Partial Class XtraReport1
        Inherits XtraReport

        Public Sub New()
            InitializeComponent()
        End Sub

        Private gr As Graphics = Graphics.FromHwnd(IntPtr.Zero)

        Private Sub xrPivotGrid1_CustomRowHeight(ByVal sender As Object, ByVal e As PivotCustomRowHeightEventArgs)
            e.RowHeight = 0
            For i As Integer = 0 To e.ColumnCount - 1
                Dim value As String = e.GetRowCellValue(i).ToString()
                Dim size As SizeF = gr.MeasureString(value, TryCast(e.DataField, XRPivotGridField).Appearance.Cell.Font, e.DataField.Width)
                Dim height As Integer = Convert.ToInt32(size.Height + 0.5)
                e.RowHeight = If(e.RowHeight > height, e.RowHeight, height)
            Next
        End Sub
    End Class
End Namespace

File: CS/XtraReport1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;
using DevExpress.XtraReports.UI.PivotGrid;

namespace WindowsFormsApplication1
{
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport
    {
        public XtraReport1()
        {
            InitializeComponent();
        }
        Graphics gr = Graphics.FromHwnd(IntPtr.Zero);
        private void xrPivotGrid1_CustomRowHeight(object sender, DevExpress.XtraReports.UI.PivotGrid.PivotCustomRowHeightEventArgs e)
        {
            e.RowHeight = 0;
            for (int i = 0; i < e.ColumnCount; i++)
            {
                string value = e.GetRowCellValue(i).ToString();
                SizeF size = gr.MeasureString(value, (e.DataField as XRPivotGridField).Appearance.Cell.Font, e.DataField.Width);
                int height = Convert.ToInt32(size.Height + 0.5);
                e.RowHeight = e.RowHeight > height?e.RowHeight : height;
            }
        }

    }
}
